html/browsers/browsing-the-web/navigating-across-documents/009.html

```diff
@@ -1,22 +1,40 @@
 <!doctype html>
-<title>Link with onclick form submit to javascript url with document.write and href navigation </title>
+<link rel="help" href="https://html.spec.whatwg.org/multipage/browsing-the-web.html#navigating-across-documents">
+<link rel="help" href="https://html.spec.whatwg.org/multipage/browsing-the-web.html#javascript-protocol">
+<link rel="help" href="https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#concept-form-submit">
+<link rel="help" href="https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#submit-get-action">
+<link rel="help" href="https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#plan-to-navigate">
+<link rel="help" href="https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#planned-navigation">
+<title>Link with onclick form submit to javascript: url with document.write and href navigation</title>
 <script src="/resources/testharness.js"></script>
 <script src="/resources/testharnessreport.js"></script>
 <div id="log"></div>
 <iframe id="test" name="test"></iframe>
-<form target="test" action="javascript:(function() {document.write('<script>parent.postMessage(&quot;write&quot;, &quot;*&quot;)</script>'); return '<script>parent.postMessage(&quot;click&quot;, &quot;*&quot;)</script>'})()"></form>
+<form target="test" action="javascript:document.write('<script>parent.postMessage(&quot;write&quot;, &quot;*&quot;)</script>');"></form>
```

Comment:
This is also significantly changing the test.  The previous test was testing both whether the postMessage from the javascript: evaluation itself happened and whether the postMessage from parsing the evaluation result happened.  The new version doesn't have an evaluation result at all.

Reply:
Some of these had to be rewritten entirely due to Chrome/Blink not honoring the `target=` when handling a "javascript: url" (which results in this test producing a blank page in Chrome). Despite that, I believe this test was wrong because the "javascript: url" code evaluation and execution is queued in a task (step 12 of [navigating-across-documents](https://html.spec.whatwg.org/multipage/browsing-the-web.html#navigating-across-documents)), so when step 11 ("Return to whatever algorithm invoked the navigation steps and continue running these steps in parallel.") returns and the algorithm moves on to navigating to href.html, it will then encounter step 6 of [navigating-across-documents](https://html.spec.whatwg.org/multipage/browsing-the-web.html#navigating-across-documents) and "Cancel any preexisting but not yet mature attempt to navigate _browsingContext_,". This is another example of under-specified behavior that I've identified, ie: there is no way to explicitly target a "preexisting ... attempt to navigate _browsingContext_," that exists within a task that's presently in a given queue (for removal at step 6)